drivers/sdhc/sdhci_common.c

```diff
@@ -437,7 +503,11 @@ static void sdhci_prepare_dma(struct sdhci_common *sdhci_ctx, struct sdhc_data *
 
 	dma_addr = z_mem_phys_addr(data->data);
```

Comment:
Should you add a check that ensures `dma_addr` is less than 32-bit, and if it isn't, use non-DMA transfer?
in case of ADMA2 we may try to use 64-bit descs mode